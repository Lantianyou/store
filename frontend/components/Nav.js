import Router from 'next/router'
import Link from 'next/link'
import NProgress from 'nprogress'
import { Mutation } from 'react-apollo'
import { TOGGLE_CART_MUTATION } from './Cart'
import Signout from './Signout'
import User from './User'
import NavStyles from './styles/NavStyles'
import CartItemCount from './CartItemCount'

Router.onRouteChangeStart = () => {
  NProgress.start()
}
Router.onRouteChangeComplete = () => {
  NProgress.done()
}
Router.onRouteChangeError = () => {
  NProgress.done()
}

const Nav = () => (
  <User>
    {({ data: { me } }) => (
      <NavStyles data-test="nav">
        <Link href="/items">
          <a>Shop</a>
        </Link>
        {me && (
          <>
            <Link href="/sell">
              <a>Sell</a>
            </Link>
            <Link href="/orders">
              <a>Orders </a>
            </Link>
            <Link href="/me">
              <a>Me</a>
            </Link>
            <Signout />
            <Mutation mutation={TOGGLE_CART_MUTATION}>
              {(toggleCart) => (
                <button onClick={toggleCart}>
                  cart
                  <CartItemCount
                    count={me.cart.reduce(
                      (tally, cartItem) => tally + cartItem.quantity,
                      0
                    )}
                  />
                </button>
              )}
            </Mutation>
          </>
        )}
        {!me && (
          <Link href="/signup">
            <a>Sign up</a>
          </Link>
        )}
      </NavStyles>
    )}
  </User>
)

export default Nav
