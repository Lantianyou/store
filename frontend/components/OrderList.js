import React, { Component } from 'react'
import { Query } from 'react-apollo'
import { formatDistance } from 'date-fns'
import Link from 'next/link'
import styled from 'styled-components'
import gql from 'graphql-tag'
import formatMoney from '../lib/formatMoney'
import OrderItemStyles from './styles/OrderItemStyles'
import Error from './ErrorMessage'

const USER_ORDERS_QUERY = gql`
  query USER_ORDERS_QUERY {
    orders(orderBy: createdAt_DESC) {
      id
      total
      createdAt
      items {
        id
        title
        price
        description
        quantity
        image
      }
    }
  }
`

const OrderUI = styled.ul`
  display: grid;
  grid-gap: 4rem;
  grid-template-columns: repeat(auto-fit, minmax(40%, 1fr));
`

class OrderList extends Component {
  render() {
    return (
      <Query query={USER_ORDERS_QUERY}>
        {({ data: { orders }, loading, error }) => {
          if (loading) {
            return <p>loading</p>
          }
          if (error) {
            return <Error error={error} />
          }
          return (
            <div>
              <h2>you have {orders.length} orders</h2>
              <OrderUI>
                {orders.map((order) => (
                  <OrderItemStyles>
                    <Link
                      href={{ pathname: '/order', query: { id: order.id } }}
                    >
                      <a>
                        <div className="order-meta">
                          <p>
                            {order.items.reduce((a, b) => a + b.quantity, 0)}
                            items
                          </p>
                          <p>{order.items.length} products</p>
                          <p>{formatDistance(order.createdAt, new Date())}</p>
                          <p>{formatMoney(order.total)}</p>
                        </div>
                        <div className="images">
                          {order.items.map((item) => (
                            <img
                              src={item.image}
                              alt={item.title}
                              key={item.id}
                            />
                          ))}
                        </div>
                      </a>
                    </Link>
                  </OrderItemStyles>
                ))}
              </OrderUI>
            </div>
          )
        }}
      </Query>
    )
  }
}

export default OrderList
