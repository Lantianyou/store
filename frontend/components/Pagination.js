import React from 'react'
import PaginationStyles from './styles/PaginationStyles'
import gql from 'graphql-tag'
import { Query } from 'react-apollo'
import Head from 'next/head'
import Link from 'next/link'
import { perPage } from '../config'

const PAGINATION_QUERY = gql`
  query PAGINATION_QUERY {
    itemsConnection {
      aggregate {
        count
      }
    }
  }
`

const Pagination = (props) => (
  <Query query={PAGINATION_QUERY}>
    {({ data, loading, error }) => {
      if (loading) {
        return <p>loading...</p>
      }
      const count = data.itemsConnection.aggregate.count
      const page = props.page
      const pages = Math.ceil(count / perPage)
      return (
        <PaginationStyles data-test="pagination">
          <Head>
            <title>
              store -{page} of {pages}
            </title>
          </Head>
          <Link
            prefetch
            href={{
              pathname: 'items',
              query: { page: page - 1 }
            }}
          >
            <a className="prev" aria-disabled={page <= 1}>
              prev
            </a>
          </Link>
          <p>{count} items total</p>
          <Link
            prefetch
            href={{
              pathname: 'items',
              query: { page: page + 1 }
            }}
          >
            <a className="next" aria-disabled={page >= pages}>
              next
            </a>
          </Link>
          <p>
            Page {props.page} of <span className="allpages">{pages}</span>
          </p>
        </PaginationStyles>
      )
    }}
  </Query>
)

export default Pagination
export { PAGINATION_QUERY }
