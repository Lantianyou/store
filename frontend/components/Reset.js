import React, { Component } from 'react'
import { Mutation } from 'react-apollo'
import gql from 'graphql-tag'
import PropTypes from 'prop-types'
import { CURRENT_USER_QUERY } from './User'
import Form from './styles/Form'
import Error from './ErrorMessage'

const REQUEST_MUTATION = gql`
  mutation REQUEST_MUTATION(
    $resetToken: String!
    $password: String!
    $confirmPassword: String!
  ) {
    resetRequest(
      resetToken: $resetToken
      password: $password
      confirmPassword: $confirmPassword
    ) {
      id
      email
      name
    }
  }
`

class Reset extends Component {
  static propTypes = {
    resetToken: PropTypes.string.isRequired
  }
  state = {
    password: '',
    confirmPassword: ''
  }
  saveToState = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }
  render() {
    return (
      <Mutation
        mutation={REQUEST_MUTATION}
        variables={{
          resetToken: this.props.resetToken,
          password: this.props.password,
          confirmPassword: this.props.confirmPassword
        }}
        refetchQueries={[
          {
            query: CURRENT_USER_QUERY
          }
        ]}
      >
        {(reset, { error, loading, called }) => {
          return (
            <Form
              method="POST"
              onSubmit={async (e) => {
                e.preventDefault()
                await reset()
                this.setState({ password: '', confirmPassword: '' })
              }}
            >
              <fieldset disabled={loading} aria-busy={loading}>
                <h2>Reset your password</h2>
                <Error error={error} />
                {!error && !loading && called && (
                  <p>Success check your email</p>
                )}
                <label htmlFor="password">
                  Email
                  <input
                    type="password"
                    name="password"
                    placeholder="password"
                    value={this.state.password}
                    onChange={this.saveToState}
                  />
                </label>
                <label htmlFor="confirmPassword">
                  Email
                  <input
                    type="password"
                    name="confirmPassword"
                    placeholder="confirm assword"
                    value={this.state.confirmPassword}
                    onChange={this.saveToState}
                  />
                </label>
                <button type="submit">Request reset</button>
              </fieldset>
            </Form>
          )
        }}
      </Mutation>
    )
  }
}
export default Reset
