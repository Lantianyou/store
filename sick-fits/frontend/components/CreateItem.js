import React, { Component } from 'react';
import Form from './styles/Form';

class CreateItem extends Component {
  state = {
    title: '',
    description: '',
    image: '',
    largeImage: '',
    price: 0,
  };

  render() {
    return (
      <Form>
        <fieldset>
          <label htmlFor="title">
            Title
            <input
              type="text"
              id="title"
              name="title"
              placeholder="title"
              required
            />
          </label>
        </fieldset>
      </Form>
    );
  }
}

export default CreateItem;
