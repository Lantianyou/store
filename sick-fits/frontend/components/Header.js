const Header = () => (
    <div>
        <div className="bar">
            <a href=""></a>
        </div>
        <div className="sub-bar">

        </div>
        <div>Cart</div>
    </div>
)