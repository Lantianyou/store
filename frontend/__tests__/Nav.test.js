import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import wait from 'waait'
import { MockedProvider } from 'react-apollo/test-utils'
import SingleItem, { SINGLE_ITEM_QUERY } from '../components/SingleItem'
import { fakeUser, fakeCartItem } from '../lib/testUtils'
import { CURRENT_USER_QUERY } from '../components/User'
import Nav from '../components/Nav'

const notSignedinMock = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: null } }
  }
]

const signedinMock = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: fakeUser() } }
  }
]

const signedinMockWithCartItem = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItem(), fakeCartItem()]
        }
      }
    }
  }
]

describe('<Nav />', () => {
  it('not signed in nav', async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedinMock}>
        <Nav />
      </MockedProvider>
    )
    await wait()
    wrapper.update()

    const nav = wrapper.find('ul[data-test="nav"]')
  })

  it('signed in nav', async () => {
    const wrapper = mount(
      <MockedProvider mocks={signedinMock}>
        <Nav />
      </MockedProvider>
    )
    await wait()
    wrapper.update()

    const nav = wrapper.find('ul[data-test="nav"]')
    expect(nav.children().length).toBe(6)
    expect(nav.text()).toContain('Sign out')
  })

  it('signed in with cart', async () => {
    const wrapper = mount(
      <MockedProvider mocks={signedinMockWithCartItem}>
        <Nav />
      </MockedProvider>
    )
    await wait()
    wrapper.update()

    const nav = wrapper.find('ul[data-test="nav"]')
    const count = nav.find('div.count')
    expect(toJSON(count)).toMatchSnapshot()
  })
})
