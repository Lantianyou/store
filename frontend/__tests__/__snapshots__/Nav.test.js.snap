// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> signed in with cart 1`] = `
<div
  className="count CartItemCount__Dot-p5gxa3-1 cCdbSw"
>
  6
</div>
`;
